product_barcode_required/models/product_barcode_mixin.py

```diff
@@ -0,0 +1,40 @@
+# Copyright 2020 Camptocamp SA
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl)
+
+from odoo import _, api, exceptions, fields, models
+
+
+class BarcodeRequiredMixin(models.AbstractModel):
+    _name = "product.barcode.required.mixin"
+    _description = "Barcode required mixin"
+
+    is_barcode_required = fields.Boolean(compute="_compute_is_barcode_required")
+
+    @api.onchange("default_code")
+    def _onchange_code(self):
+        for rec in self:
+            if rec.is_barcode_required and rec.default_code:
+                rec.barcode = rec.default_code
+
+    @api.depends("type", "barcode")
+    def _compute_is_barcode_required(self):
+        for rec in self:
+            rec.is_barcode_required = (
+                rec._is_barcode_required_enabled() and rec._is_barcode_required()
+            )
+
+    def _is_barcode_required(self):
```

Comment:
I would prefer an other name for this method.... `_is_barcode_missing`

Reply:
renamed